stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/StagGenerator.java

```diff
@@ -102,111 +95,50 @@ public void generateTypeAdapters() throws IOException {
-    private static MethodSpec getReadFromAdapterSpec(@NotNull TypeVariableName genericTypeName) {
-        return MethodSpec.methodBuilder("readFromAdapter")
-                .addModifiers(Modifier.STATIC)
-                .returns(genericTypeName)
-                .addTypeVariable(genericTypeName)
-                .addException(IOException.class)
-                .addParameter(Gson.class, "gson")
-                .addParameter(ParameterizedTypeName.get(ClassName.get(Class.class), genericTypeName), "clazz")
-                .addParameter(JsonReader.class, "in")
-                .addCode("return gson.getAdapter(clazz).read(in);\n")
-                .build();
-    }
-
-    @NotNull
-    private static MethodSpec getWriteListToAdapterSpec(@NotNull TypeVariableName genericTypeName) {
-        return MethodSpec.methodBuilder("writeListToAdapter")
-                .addModifiers(Modifier.STATIC)
-                .returns(void.class)
-                .addTypeVariable(genericTypeName)
-                .addException(IOException.class)
-                .addParameter(Gson.class, "gson")
-                .addParameter(ParameterizedTypeName.get(ClassName.get(Class.class), genericTypeName), "clazz")
-                .addParameter(JsonWriter.class, "out")
-                .addParameter(ParameterizedTypeName.get(ClassName.get(ArrayList.class), genericTypeName),
-                              "list")
-                .addCode("com.google.gson.TypeAdapter<T> typeAdapter = gson.getAdapter(clazz);\n" +
-                         '\n' +
-                         "for (T object : list) {\n" +
-                         "\ttypeAdapter.write(out, object);\n" +
-                         "}\n")
-                .build();
-    }
-
-    @NotNull
-    private static MethodSpec getReadListFromAdapterSpec(@NotNull TypeVariableName genericTypeName) {
-        return MethodSpec.methodBuilder("readListFromAdapter")
-                .addModifiers(Modifier.STATIC)
-                .returns(ParameterizedTypeName.get(ClassName.get(ArrayList.class), genericTypeName))
-                .addTypeVariable(genericTypeName)
-                .addException(IOException.class)
-                .addParameter(Gson.class, "gson")
-                .addParameter(ParameterizedTypeName.get(ClassName.get(Class.class), genericTypeName), "clazz")
-                .addParameter(JsonReader.class, "in")
-                .addCode("ArrayList<T> list = new java.util.ArrayList<>();\n" +
-                         "com.google.gson.TypeAdapter<T> typeAdapter = gson.getAdapter(clazz);\n" +
-                         '\n' +
-                         "while(in.hasNext()){\n" +
-                         "\tlist.add(typeAdapter.read(in));\n" +
-                         "}\n" +
-                         '\n' +
-                         "return list;\n")
-                .build();
-    }
-
-    @NotNull
-    private static TypeSpec getAdapterFactorySpec(Set<Element> types) {
+    private TypeSpec getAdapterFactorySpec() {
```

Comment:
My personal preference is to take SRP to its limit and keep every individual class fully
separated.  The TypeAdapter and TypeAdapterFactory have different purposes that can
be composed in the consuming application in various ways.  If the two are collapsed
together then this flexibility starts to diminish and the consumer is boxed into a corner.

The second option would make the class instances stateful which would be more difficult
to understand as a casual observer.  Also, care would need to be taken with respect to
the memory model in this scenario as - I believe - Gson can initialize type adapters in
separate threads.

Not my call though. 😄